src/main/java/bio/overture/ego/security/OAuth2SsoFilter.java

```diff
@@ -192,4 +200,23 @@ public FacebookFilter(OAuth2ClientResources client) {
               super.restTemplate));
     }
   }
+
+  class OrcidFilter extends OAuth2SsoChildFilter {
+    public OrcidFilter(OAuth2ClientResources client) {
+      super("/oauth/login/orcid", client);
+      super.setTokenServices(
+          new OAuth2UserInfoTokenServices(
+              client.getResource().getUserInfoUri(),
+              client.getClient().getClientId(),
+              super.restTemplate) {
+            @Override
+            protected Map<String, Object> transformMap(
+                Map<String, Object> map, String accessToken) {
+              val orcid = map.get("sub").toString();
```

Comment:
this should be renamed to `subject`. Also, need to add error handling for when the subject is NOT orchid

Reply:
the `map` returned here from the `userInfoUri` comes back in this structure:
`{id=https://orcid.org/{orcid}, sub={orcid}, name=null, family_name=string, given_name=string}`
Mentioned here: https://github.com/ORCID/ORCID-Source/blob/master/orcid-web/ORCID_AUTH_WITH_OPENID_CONNECT.md#authorization-code-flow
So the `sub` value will always be the orcid. Not sure if that's what you're referring to in `add error handling for when the subject is NOT orcid` ?

Reply:
Just being defensive. Add a simple check, and if it fails, return a 502 bad gateway error, indicating an invalid response from the upstream (orchid) server, since you were expecting "sub" to be "orcid" but was something else instead